TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Command/NodeCommandControllerPlugin.php

```diff
@@ -856,6 +865,95 @@ protected function removeNode($nodeIdentifier, $dimensionsHash)
     }
 
     /**
+     * Remove nodes with invalid dimension values
+     *
+     * This removes nodes which have dimension values not fitting to the current dimension configuration
+     *
+     * @param NodeType $nodeType Only for this node type, if specified
+     * @param string $workspaceName Name of the workspace to consider
+     * @param boolean $dryRun Simulate?
+     * @return void
+     */
+    public function removeNodesWithInvalidDimensions(NodeType $nodeType, $workspaceName, $dryRun)
+    {
+        $allowedDimensionCombinations = $this->contentDimensionCombinator->getAllAllowedCombinations();
+
+        if ($nodeType !== null) {
+            $this->output->outputLine('Checking nodes of type "%s" for invalid dimensions ...', array($nodeType->getName()));
+            $nodesArray = $this->collectNodesWithInvalidDimensions($nodeType, $workspaceName, $allowedDimensionCombinations);
+        } else {
```

Comment:
Just realized this case doesn't exist, as the signature of the method doesn't even allow giving `null` for the $nodeType... IMHO either we throw this code away or rework the signature. I mean actually you shouldn't need a node type to remove nodes with invalid dimensions. And looking at the call it seems NodeType can be `null` so the signature has to be reworked to make nodeType an optional argument.